### PR TITLE
Social: Add simple JS React test

### PR DESCRIPTION
**Pull request by brbrr**
Adds two simple JS React tests:
- for Social plugin
- for publicize-components js-package

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

pd5faL-jc-p2
#### Does this pull request change what data or activity we track or use?

n/a
#### Testing instructions:

* Make sure it runs locally and in CI
*

### CODE REVIEW

File: projects/plugins/social/src/js/components/admin-page/test/index.test.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { render, screen } from '@testing-library/react';
+import { renderHook } from '@testing-library/react-hooks';
+import { useSelect, createReduxStore, register } from '@wordpress/data';
+import React from 'react';
+import { STORE_ID, storeConfig } from '../../../store';
+import Admin from '../index';
+
+const store = createReduxStore( STORE_ID, storeConfig );
+register( store );
```
Comment (brbrr) [inline projects/plugins/social/src/js/components/admin-page/test/index.test.jsx:9]: This is a bit ugly and probably worth exporting into a jest setup file. Alternatively, we can update the store init to be in line with other projects in monorepo: https://github.com/Automattic/jetpack/blob/trunk/projects/js-packages/connection/state/store.jsx#L19-L27

File: projects/plugins/social/src/js/components/admin-page/test/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+describe( 'load the app', () => {
+	test( 'container renders', () => {
+		const version = '99.9';
+		let storeSelect;
+		renderHook( () => useSelect( select => ( storeSelect = select( STORE_ID ) ) ) );
+		jest.spyOn( storeSelect, 'showPricingPage' ).mockReset().mockReturnValue( true );
+		jest.spyOn( storeSelect, 'hasPaidPlan' ).mockReset().mockReturnValue( true );
+		jest.spyOn( storeSelect, 'isShareLimitEnabled' ).mockReset().mockReturnValue( true );
```
Comment (brbrr) [inline projects/plugins/social/src/js/components/admin-page/test/index.test.jsx:18]: These calls are not particularly necessary.

File: projects/plugins/social/src/js/components/admin-page/test/index.test.jsx (hunk 1 above)
Comment (pablinos) [reply projects/plugins/social/src/js/components/admin-page/test/index.test.jsx:9]: > Alternatively, we can update the store init to be in line with other projects in monorepo

I think that's the way to go. Do you think we should do that first or jest get this merged?

File: projects/plugins/social/src/js/components/admin-page/test/index.test.jsx (hunk 2 above)
Comment (pablinos) [reply projects/plugins/social/src/js/components/admin-page/test/index.test.jsx:18]: Do you mean that they're just examples?

**Review by pablinos — COMMENTED** [review]
This looks great to me @brbrr, thanks very much for working on it. This is definitely something that we can build on.

File: projects/plugins/social/src/js/components/admin-page/test/index.test.jsx (hunk 1 above)
Comment (brbrr) [reply projects/plugins/social/src/js/components/admin-page/test/index.test.jsx:9]: I'm fine merging it as is and then aligning the store registration in follow-up PR.

File: projects/plugins/social/src/js/components/admin-page/test/index.test.jsx (hunk 2 above)
Comment (brbrr) [reply projects/plugins/social/src/js/components/admin-page/test/index.test.jsx:18]: Yeah, kinda. I added it there while I was trying to figure out why the store was empty. Turns out it wasn't registered automatically, as in other plugins. Also, the above store keys are populated with default values, so there is no need to explicitly set them via `spyOn` calls in this "default" rendering case.

**Review by adimoldovan — APPROVED** [review]

**Pull request merged**